src/aws/ecr_helpers.go

```diff
@@ -138,14 +143,25 @@ func buildImageName(dockerConfig types.DockerConfig, dockerComposeProjectName, s
 }
 
 // returns an image with version tag if applicable. uses the application version otherwise
-func getRepositoryConfig(deployConfig types.DeployConfig, imageName string) (string, string) {
+func getRepositoryConfig(deployConfig types.DeployConfig, serviceLocation, imageName string) (string, string, error) {
 	config := strings.Split(imageName, ":")
 	repositoryName := config[0]
 	var version string
+	var err error
 	if len(config) > 1 {
 		version = config[1]
 	} else {
-		version = deployConfig.AppConfig.Version
+		version, err = getCommitSHA(deployConfig, serviceLocation)
+	}
+	return repositoryName, version, err
+}
+
+func getCommitSHA(deployConfig types.DeployConfig, serviceLocation string) (string, error) {
+	cmd := exec.Command("git", "rev-list", "-1", "HEAD", serviceLocation)
+	cmd.Dir = deployConfig.AppDir
+	output, err := cmd.Output()
+	if err != nil {
+		return "", err
 	}
-	return repositoryName, version
+	return strings.Trim(string(output), "\n"), err
```

Comment:
You can use `strings.TrimSpace` here. Also please use `return ..., nil` to make it more clear we are not returning an error here